patches/minecraft/net/minecraft/block/RedstoneWireBlock.java.patch

```diff
@@ -27,10 +50,12 @@
 -   }
 -
 -   protected static boolean func_176343_a(BlockState p_176343_0_, @Nullable Direction p_176343_1_) {
+-      if (p_176343_0_.func_203425_a(Blocks.field_150488_af)) {
 +   protected static boolean canConnectTo(BlockState p_176343_0_, IBlockReader world, BlockPos pos, @Nullable Direction p_176343_1_) {
-       if (p_176343_0_.func_203425_a(Blocks.field_150488_af)) {
++      if (p_176343_0_.func_235714_a_(net.minecraftforge.common.Tags.Blocks.REDSTONE_WIRE)) {
```

Comment:
A tag is not valid for this. Using a tag means people could add random blocks to the redstone tag, which a vanilla client would have no idea of, and thus would cause a desync. This should only consider blocks that extend RedstoneWireBlock.

On top of that, I have a concern about this: what about all the blocks that want to implement "non-connecting" redstone? Like other colors of redstone?

Reply:
1. I got your point, I will remove the tags and change it to `instanceof`

2. What about adding a method like `boolean canConnectToOtherRedstone()` in `RedstoneWireBlock` which returns `false` be default. Then for every check:
```java
if(blockState.isIn(Blocks.REDSTONE_WIRE){
  return true;
}
// I do the check here, so if the method returns false, it will still go through the other if branches
else if(blockState.getBlock() instanceof RedstoneWireBlock && ((RedstoneWireBlock) blockState.getBlock()).canConnectToOtherRedstone()){
  return true;
}else if {
// ...
}
```
This won't cause any breaking changes with existing mods, since the behavior will only change if they override the method and change the return value.

Reply:
I like that idea, it sounds like it would work fine with mods that currently add non-connecting variants of redstone wire. Example: [More Wires](https://github.com/mrp-v2/MoreWires)

Reply:
This sounds like a better option to me, however, the blocks would probably need to know the other block wants *them* specificall. Something like like:
```java
if(otherblock == thisblock || (otherblock instanceof RedstoneWireBlock && otherblock.canConnectToOther(this) && this.canConnectToOther(otherblock)))
```
Also, if you want this in 1.16.x, we need to make sure the default state doesn't break semantics. So by default canConnectToOther should behave the same way the old logic would have worked.

Reply:
If we want it to behave as before, you probably want an or condition between the two can connect calls. Or maybe make some sort of 3 way return for block, allow, and default.

Reply:
I think what giga suggested is already fine. If it was an or condition, then redstones that don't want to connect to other redstones would be forced to connect to restones that do want to connect to other redstones. I also think a 3-way return is uneccesary, as returning default would probably be no different from returning allow, unless I'm misunderstanding what you mean.